### ISSUE TITLE
Custom HitTestBehavior for Scrollables

### ISSUE THREAD

**Issue by gertdrui** [flutter/flutter#76525]
## Use case

I've needed to have full screen scrollables - either PageView or CustomScrollView - as the frontmost widget in a Stack that also need to allow passing gestures through to the widgets behind, for example when there is a gap between the elements in the CustomScrollView. 

With the options currently available, this is not possible, since all scrollviews use Scrollable, which is hardcoded as `HitTestBehavior.opaque`.

Currently to make this work I've had to copy the contents of Scrollable and ScrollView or PageView to make my own version just so I could set the HitTest behavior to translucent.

## Proposal

I propose to make HitTestBehavior configurable in Scrollable as well as the built-in ScrollViews and PageViews that use it.

**Comment by HansMuller** [comment]
CC @Piinks 

This may have unintended consequences. When more than one overlapping gesture detector tries to recognize gestures that are not completely distinct, the results can be confusing and not what you want.

**Comment by nt4f04uNd** [comment]
fyi similar proposal https://github.com/flutter/flutter/issues/75099

**Comment by Hixie** [comment]
@gertdrui Can you elaborate on your use case? It seems weird, as @HansMuller says, to have taps go "through" a scrolling widget to non-scrolling widgets below it. Maybe a little video showing the UI you have in mind, if you have it built already?

**Comment by gertdrui** [comment]
@nt4f04uNd Setting it to HitTestBehavior.translucent does work for what I'm trying to achieve because I'm only interested in taps reaching the widgets underneath while dragging is captured by the scrollview.

@Hixie 
Here's my working example, the elements in the background are interactable while dragging anywhere on the screen controls the scrollable area.
[video](https://www.youtube.com/watch?v=WI5kTDGDIPk&feature=youtu.be)

**Comment by Hixie** [comment]
Ohhhh, I see. It's "above" the list that's static (in the Y direction, when the list is at zero). Interesting.

One way to do that today would be to have a special sliver that never scrolls, always reports a scroll height of zero, and paints across the entire list, FWIW.

But I see what you're suggesting. It sounds like what you want is a widget like IgnorePointer/AbsorbPointer that answers yes, yes, no to the questions in https://github.com/flutter/flutter/issues/74733#issuecomment-767859584.

That should be relatively straightforward to add. The trickiest part would be documenting it in a way that people understand...

**Comment by Hixie** [comment]
Actually the trickiest part might be _naming_ the widget.

**Comment by nt4f04uNd** [comment]
> It sounds like what you want is a widget like IgnorePointer/AbsorbPointer that answers yes, yes, no to the questions in [#74733 (comment)](https://github.com/flutter/flutter/issues/74733#issuecomment-767859584).

would this be done instead of #75099, do you think it's better? i think it's a bit tricky and even with decent explanations in docs people will still open issues where they don't understand what's happening

is it bad idea to hit test all the Stack for instance? or to define any other rules that would allow this to be handled by the framework by default and would not require any additional steps from devs

**Comment by gertdrui** [comment]
> Actually the trickiest part might be _naming_ the widget.

yes, yes, no would be the exact opposite of absorb so I guess that would be exude :)

**Comment by Hixie** [comment]
The problem with having the Scrollable configurable to translucent is that it would mean that if you put _any_ render object widget between the Stack and the Scrollable, e.g. a SizedBox, Container, anything, it would no longer do what you want. I'm not even sure that ListView itself doesn't put a render object widget between its parent and the gesture detector that does the scrolling...

**Comment by nt4f04uNd** [comment]
@Hixie did you miss the https://github.com/flutter/flutter/issues/76525#issuecomment-783801310?

**Comment by Hixie** [comment]
No, but I didn't have an answer that wouldn't involve doing a lot of work, so I hadn't answered yet. :-) Whoever fixes this bug is probably the person who will have to do the research to figure out the answers.

**Comment by nmfisher** [comment]
From what I've pieced together across a few different issues, it's currently impossible to trigger both "down1" and "down2" with the following:
```
Stack(children:[
Listener(onPointerDown:(_) {
print("down1")
},child:Container()),
Listener(onPointerDown:(_) {
print("down2")
},child:Container()),
]);
```

And this issue would solve that.

If so, then +1 from me.
